spark_pipeline_framework/transformers/framework_fill_na_transformer/v1/framework_fill_na_transformer.py

```diff
@@ -0,0 +1,101 @@
+    ProgressLogMetric,
+)
+from spark_pipeline_framework.progress_logger.progress_logger import ProgressLogger
+from spark_pipeline_framework.transformers.framework_transformer.v1.framework_transformer import (
+    FrameworkTransformer,
+)
+
+
+def get_dtype(df: DataFrame, colnames: List[str]) -> Dict[str, str]:
+    return dict(df.select(colnames).dtypes)
+
+
+class FrameworkFillNaTransformer(FrameworkTransformer):
+    """
+    Replace NA/Null values with a specified replacement_value for specified columns_to_check for null values
+    """
+
+    # noinspection PyUnusedLocal
+    @keyword_only
+    def __init__(
+        self,
+        view: str,
+        replacement_value: Any,
+        columns_to_check: List[str],
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+    ):
+        super().__init__()
+        self.logger = get_logger(__name__)
+
+        self.view: Param = Param(self, "view", "")
+        self._setDefault(view=view)
+
+        self.replacement_value: Param = Param(self, "replacement_value", "")
+        self._setDefault(replacement_value=replacement_value)
+
+        self.columns_to_check: Param = Param(self, "columns_to_check", "")
+        self._setDefault(columns_to_check=columns_to_check)
+
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    @keyword_only
+    def setParams(
+        self,
+        view: str,
+        replacement_value: Any,
+        columns_to_check: List[str],
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+    ) -> Any:
+        kwargs = self._input_kwargs
+        super().setParams(
+            name=name, parameters=parameters, progress_logger=progress_logger
+        )
+        return self._set(**kwargs)
+
+    def _transform(self, df: DataFrame) -> DataFrame:
+        columns_to_fill: List[str] = self.getColumnsToCheck()
+        replacement_value: Any = self.getReplacementValue()
+        view: str = self.getView()
+        progress_logger: Optional[ProgressLogger] = self.getProgressLogger()
+
+        with ProgressLogMetric(name=f"{view}_fill_na", progress_logger=progress_logger):
+            self.logger.info(
+                f"filling rows if any null values with replacement_value found for columns: {columns_to_fill}"
+            )
+            df_with_na: DataFrame = df.sql_ctx.table(view)
+            data_types = get_dtype(df_with_na, columns_to_fill)
+            for col in columns_to_fill:
+                if data_types[col] == "string":
+                    df_with_filled_na = df_with_na.na.replace(
+                        "", value=replacement_value, subset=col
+                    )
+                else:
+                    df_with_filled_na = df_with_na.na.fill(
+                        value=replacement_value, subset=col
+                    )
```

Comment:
This looks like there might be an error if there is more than one column defined, since `df_with_filled_na` is defined with each loop. Can you confirm that this is not the case? I'm not seeing how it wouldn't be.

Reply:
You're right. I'll make the adjustments.